From 81e1cad6da0baa6c53e25834508229a069678e90 Mon Sep 17 00:00:00 2001
From: Scott K Logan <logans@cottsay.net>
Date: Wed, 14 Feb 2024 14:13:35 -0600
Subject: [PATCH] Explicitly specify Python 3 interpreter

A recent change to RHEL 8 results in the Python 3.11 interpreter being
present during this package build due to an indirect dependency. The
find_package(Python3) search is discovering the newer interpreter and
attempting to use it instead of the system's default interpreter, for
which we've installed other dependencies.

This flag should force the find_package logic to use the system's
default interpreter as it was doing before.
---
 rpm/ros-humble-pcl-conversions.spec | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rpm/ros-humble-pcl-conversions.spec b/rpm/ros-humble-pcl-conversions.spec
index bb029b10a9fb..8ddf5f3c7cfd 100644
--- a/rpm/ros-humble-pcl-conversions.spec
+++ b/rpm/ros-humble-pcl-conversions.spec
@@ -64,6 +64,7 @@ mkdir -p .obj-%{_target_platform} && cd .obj-%{_target_platform}
 %if !0%{?with_tests}
     -DBUILD_TESTING=OFF \
 %endif
+    -DPython3_EXECUTABLE="%{__python3}" \
     ..
 
 %make_build
-- 
2.43.0

